src/app/clusters/concentration-measurement-server/concentration-measurement-server.h

```diff
@@ -233,35 +279,37 @@ class Instance : public AttributeAccessInterface
     };
 
     /**
-     * This checks is a given value is greater than a given value.
+     * This checks if a given nullable float is less than or equal to another given nullable float.
      * @param value The value to check.
-     * @param valueToBeGreaterThan The value to be greater than.
-     * @return true if the value is greater than the given value.
+     * @param valueToBeLessThanOrEqualTo The value to be less than or equal to.
+     * @return true if value is less than or equal to valueToBeLessThanOrEqualTo, or if either of the values is Null.
      */
-    static bool CheckConstraintsGreaterThan(DataModel::Nullable<float> value, DataModel::Nullable<float> valueToBeGreaterThan)
+    static bool CheckConstraintsLessThanOrEqualTo(DataModel::Nullable<float> value,
+                                                  DataModel::Nullable<float> valueToBeLessThanOrEqualTo)
     {
-        if (!valueToBeGreaterThan.IsNull() && !value.IsNull() && (valueToBeGreaterThan.Value() > value.Value()))
+        if (valueToBeLessThanOrEqualTo.IsNull() || value.IsNull() || (value.Value() <= valueToBeLessThanOrEqualTo.Value()))
```

Comment:
nit: `if (foo) { return true; }  return false` seems to just be a longer version of `return foo`. 

This is throughout here.

if there is a reason to split for readability, then split should be non-one-like. so either:

```
   return valueToBeLessThanOrEqualTo.IsNull() || value.IsNull() || (value.Value() <= valueToBeLessThanOrEqualTo.Value())
```

or if this is more readable:

```
if (valueToBeLessThanOrEqualTo.IsNull() || value.IsNull())
{
    return false;
}

return value.Value() <= valueToBeLessThanOrEqualTo.Value();
```

if "input validation" vs "actual check" is important for readability.